modules/eks/submodules/irsa/variables.tf

```diff
@@ -0,0 +1,128 @@
+
+variable "eks_info" {
+  description = <<EOF
+    cluster = {
+      addons            = List of addons
+      specs             = Cluster spes. {
+        name                      = Cluster name.
+        endpoint                  = Cluster endpont.
+        kubernetes_network_config = Cluster k8s nw config.
+      }
+      version           = K8s version.
+      arn               = EKS Cluster arn.
+      security_group_id = EKS Cluster security group id.
+      endpoint          = EKS Cluster API endpoint.
+      roles             = Default IAM Roles associated with the EKS cluster. {
+        name = string
+        arn = string
+      }
+      custom_roles      = Custom IAM Roles associated with the EKS cluster. {
+        rolearn  = string
+        username = string
+        groups   = list(string)
+      }
+      oidc = {
+        arn = OIDC provider ARN.
+        url = OIDC provider url.
+        cert = {
+          thumbprint_list = OIDC cert thumbprints.
+          url             = OIDC cert URL.
+      }
+    }
+    nodes = {
+      security_group_id = EKS Nodes security group id.
+      roles = IAM Roles associated with the EKS Nodes.{
+        name = string
+        arn  = string
+      }
+    }
+    kubeconfig = Kubeconfig details.{
+      path       = string
+      extra_args = string
+    }
+  EOF
+  type = object({
+    k8s_pre_setup_sh_file = string
+    cluster = object({
+      addons = list(string)
+      specs = object({
+        name                      = string
+        endpoint                  = string
+        kubernetes_network_config = list(map(any))
+        certificate_authority     = list(map(any))
+      })
+      version           = string
+      arn               = string
+      security_group_id = string
+      endpoint          = string
+      roles = list(object({
+        name = string
+        arn  = string
+      }))
+      custom_roles = list(object({
+        rolearn  = string
+        username = string
+        groups   = list(string)
+      }))
+      oidc = object({
+        arn = string
+        url = string
+        cert = object({
+          thumbprint_list = list(string)
+          url             = string
+
+        })
+      })
+    })
+    nodes = object({
+      security_group_id = string
+      roles = list(object({
+        name = string
+        arn  = string
+      }))
+    })
+    kubeconfig = object({
+      path       = string
+      extra_args = string
+    })
+  })
+}
+
+variable "use_cluster_odc_idp" {
+  description = <<EOF
+    Toogle to uset the oidc idp connector in the trust policy.
+    Set to `true` if the cluster and the hosted zone are in different aws accounts.
+  EOF
+  type        = bool
+  default     = true
+}
+
+variable "external_dns" {
+  description = "Config to enable irsa for external-dns"
+
+  type = object({
+    enabled             = optional(bool, false)
```

Comment:
nillable instead?

Reply:
same as before, I prefer to be explicit rather than condition based on null which causes issues like the bastion

Reply:
I think its slightly different because of how the data is used and then later referenced. This detail is a leaf not something that is created and used elsewhere.

Reply:
I do not think nillables are preferred for this purpose in terraform, nor do I see what we gain.